sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts

```diff
@@ -332,24 +336,43 @@ export class LocalTxnDataSerializer implements TxnDataSerializer {
       gasBudget: originalTx.data.gasBudget,
       sender: signerAddress,
     };
-
-    return await this.serializeTransactionData(txData);
+    const version = await this.provider.getRpcApiVersion();
+    const format = shouldUseOldSharedObjectAPI(version)
+    ? 'TransactionData_Deprecated'
+    : 'TransactionData';
+    
+    if (shouldUseIntentSigning(version)) {
+      return await this.serializeIntentTransactionData(txData, format);
+    } else {
+      return await this.serializeTransactionData(txData, format);
+    }
   }
 
   private async serializeTransactionData(
     tx: TransactionData,
+    format: string,
     // TODO: derive the buffer size automatically
-    size: number = 8192
+    size: number = 8192,
   ): Promise<Base64DataBuffer> {
-    const version = await this.provider.getRpcApiVersion();
-    const format = shouldUseOldSharedObjectAPI(version)
-      ? 'TransactionData_Deprecated'
-      : 'TransactionData';
-
     const dataBytes = bcs.ser(format, tx, size).toBytes();
     const serialized = new Uint8Array(TYPE_TAG.length + dataBytes.length);
     serialized.set(TYPE_TAG);
     serialized.set(dataBytes, TYPE_TAG.length);
     return new Base64DataBuffer(serialized);
   }
+
+  async serializeIntentTransactionData(
+    tx: TransactionData,
+    format: string,
+    size: number = 8192,
+  ): Promise<Base64DataBuffer> {
+    const dataBytes = bcs.ser(format, tx, size).toBytes();
+    const serialized = new Uint8Array(INTENT_BYTES.length + dataBytes.length);
```

Comment:
I noticed that `TYPE_TAG.length` is removed from here, could you point me to the corresponding rust change?

Reply:
https://github.com/MystenLabs/sui/pull/5032/files#diff-80bf625d87d89549275351d95cfdfab4a6c2a1311804adbc5f1a7fcff225f049L2481 here we parse the bytes after the INTENT_BYTES in validator (the previously used from_signable_bytes in fact throw out the TYPE_TAG, which is kind of redundant to include it when serializing)